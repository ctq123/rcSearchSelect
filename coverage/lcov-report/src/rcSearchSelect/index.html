<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/rcSearchSelect</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> src/rcSearchSelect
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">37.58% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>56/149</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">31.58% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>42/133</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">48.39% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>15/31</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">38.36% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>56/146</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="ClickOut.jsx"><a href="ClickOut.jsx.html">ClickOut.jsx</a></td>
	<td data-value="31.82" class="pic low"><div class="chart"><div class="cover-fill" style="width: 31%;"></div><div class="cover-empty" style="width:69%;"></div></div></td>
	<td data-value="31.82" class="pct low">31.82%</td>
	<td data-value="22" class="abs low">7/22</td>
	<td data-value="8.33" class="pct low">8.33%</td>
	<td data-value="12" class="abs low">1/12</td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="7" class="abs low">2/7</td>
	<td data-value="35" class="pct low">35%</td>
	<td data-value="20" class="abs low">7/20</td>
	</tr>

<tr>
	<td class="file low" data-value="SearchSelect.jsx"><a href="SearchSelect.jsx.html">SearchSelect.jsx</a></td>
	<td data-value="38.58" class="pic low"><div class="chart"><div class="cover-fill" style="width: 38%;"></div><div class="cover-empty" style="width:62%;"></div></div></td>
	<td data-value="38.58" class="pct low">38.58%</td>
	<td data-value="127" class="abs low">49/127</td>
	<td data-value="33.88" class="pct low">33.88%</td>
	<td data-value="121" class="abs low">41/121</td>
	<td data-value="54.17" class="pct medium">54.17%</td>
	<td data-value="24" class="abs medium">13/24</td>
	<td data-value="38.89" class="pct low">38.89%</td>
	<td data-value="126" class="abs low">49/126</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Nov 05 2018 11:08:20 GMT+0800 (中国标准时间)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
