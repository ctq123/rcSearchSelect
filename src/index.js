export {default as SearchSelect} from './rcSearchSelect/SearchSelect'