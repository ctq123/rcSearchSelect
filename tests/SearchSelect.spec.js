import React from 'react'
import { mount, shallow, render } from 'enzyme'
import { SearchSelect } from '../src'

describe('SelectSelect unitTest',  ()=> {
  const dataSource = [{id:1, name: 'Alan', age: 19}, {id:2, name: 'Carry', age: 20}]

  it('renders correctly', () => {
    const wrapper = render(<SearchSelect dataSource={dataSource} keyField='id' labelField='name' />)

    expect(wrapper).toMatchSnapshot()
  })

  it('set defaultValue based on value', () => {
    const defaultValue = {id:2, name: 'Carry', age: 20}
    const wrapper = shallow(<SearchSelect dataSource={dataSource} keyField='id' labelField='name' defaultValue={defaultValue} />)

    expect(wrapper.state().inputVal).toBe('Carry')
  })

  it('placehoder', () => {
    const placeholder = 'input text'
    const wrapper = shallow(<SearchSelect dataSource={dataSource} keyField='id' labelField='name' placeholder={placeholder} />)
    const inputNode = wrapper.find('input')

    expect(wrapper.state().inputVal).toBe('')
    expect(inputNode.prop('placeholder')).toEqual(placeholder)
  })

  
  it('set dropdwonHeight', () => {
    const dropdwonHeight = 30
    const wrapper = shallow(<SearchSelect dataSource={dataSource} keyField='id' labelField='name' dropdwonHeight={dropdwonHeight} />)
    wrapper.find('input').simulate('focus')
    const node = wrapper.find('.dropdown .dropdwon-placement')

    expect(node.prop('style').height).toEqual(30)
  })

  it('set direction', () => {
    const wrapper = shallow(<SearchSelect dataSource={dataSource} keyField='id' labelField='name' />)
    expect(wrapper.find('.dropdown-parent').length).toEqual(1)

    wrapper.setProps({ direction: 'up' })
    expect(wrapper.find('.dropup-parent').length).toEqual(1)
  })


  it('set inputValue from extranal data', () => {
    const wrapper = shallow(<SearchSelect dataSource={dataSource} keyField='id' labelField='name' />)
    
    let exdata = {id: 1, label: 'Alan'}
    wrapper.setProps({ setValueObj: exdata })
    expect(wrapper.state().inputVal).toBe('Alan')
    
    exdata = {id: 2, label: null}
    wrapper.setProps({ setValueObj: exdata })
    expect(wrapper.state().inputVal).toBe('')

    exdata = {id: 3, label: 'hahaha'}
    wrapper.setProps({ setValueObj: exdata })
    expect(wrapper.state().inputVal).toBe('')

    exdata = {id: 4, label: 'Carry'}
    wrapper.setProps({ setValueObj: exdata })
    expect(wrapper.state().inputVal).toBe('Carry')
  })

  it('set inputValue when user input text', () => {
    const wrapperM = mount(<SearchSelect dataSource={dataSource} keyField='id' labelField='name' />)
    wrapperM.find('input').simulate('focus')
    wrapperM.find('input').simulate('change', { target: { value: 'Car' } })
    expect(wrapperM.find('input').prop('value')).toEqual('Car')
  })

})