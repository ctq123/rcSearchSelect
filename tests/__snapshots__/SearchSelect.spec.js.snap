// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectSelect unitTest renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "rc-searchselect",
    },
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "attribs": Object {
              "class": "dropdown-parent",
            },
            "children": Array [
              Object {
                "attribs": Object {},
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "input-con",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "auto-input rc-input",
                "placeholder": "",
                "value": "",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "placeholder": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "placeholder": undefined,
                "value": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      Object {
        "attribs": Object {
          "class": "input-con",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "auto-input rc-input",
              "placeholder": "",
              "value": "",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "placeholder": undefined,
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "placeholder": undefined,
              "value": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "attribs": Object {
                "class": "dropdown-parent",
              },
              "children": Array [
                Object {
                  "attribs": Object {},
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {},
                  "x-attribsPrefix": Object {},
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
